kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyInitializer.java

```diff
@@ -94,9 +100,13 @@ public void initChannel(SocketChannel ch) {
         }
     }
 
+    private static void addBeforeErrorHandler(ChannelPipeline pipeline, String name, ChannelHandler handler) {
```

Comment:
Why do you take this approach to building up the chain?  Can't you just add the error handler last to get the same effect?

Reply:
The issue is we need to configure the pipeline with just the SNIHandler and error handler, then add all the other handlers to the pipeline later after the binding is resolved.

Maybe after resolution it would be cleaner to remove the error handler, add the rest of the handlers, then add errorhandler last.